packages/calcite-components/src/components/input-number/input-number.tsx

```diff
@@ -840,12 +841,19 @@ export class InputNumber
       useGrouping: this.groupSeparator
     };
 
-    const sanitizedValue = sanitizeNumberString(
-      // no need to delocalize a string that ia already in latn numerals
-      (this.numberingSystem && this.numberingSystem !== "latn") || defaultNumberingSystem !== "latn"
-        ? numberStringFormatter.delocalize(value)
-        : value
-    );
+    const isValueDeleted =
+      this.previousValue?.length > value.length || this.value?.length > value.length;
+    const hasTrailingDecimalSeparator = value.charAt(value.length - 1) === ".";
```

Comment:
I think the `.` should be `numberFormatter.decimal` here, since it is being done before delocalization

Reply:
i don't think so, if you see that `setNumberValue` parses delocalized value in `inputNumberInputHandler`. The only instance when we try to check if there is a trailingDecimalSeparator is when the value is deleted which invokes ` inputNumberInputHandler` method.